components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -412,8 +415,13 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
 
             AuthenticatedUser authorizedUser = tokReqMsgCtx.getAuthorizedUser();
             if (authorizedUser.getAuthenticatedSubjectIdentifier() == null) {
-                authorizedUser.setAuthenticatedSubjectIdentifier(
-                        getSubjectClaim(getServiceProvider(tokReqMsgCtx.getOauth2AccessTokenReqDTO()), authorizedUser));
+                if (isOfTypeApplicationUser || !useClientIdAsSubClaimForAppTokensEnabled) {
+                    authorizedUser.setAuthenticatedSubjectIdentifier(
+                            getSubjectClaim(getServiceProvider(tokReqMsgCtx.getOauth2AccessTokenReqDTO()),
+                                    authorizedUser));
+                } else {
+                    authorizedUser.setAuthenticatedSubjectIdentifier(oAuthAppDO.getOauthConsumerKey());
+                }
```

Comment:
Would be much readable if we write the condition as,

isOfTypeApplicationUser && useClientIdAsSubClaimForAppTokensEnabled

rather than using a negative ||

WDYT?

Reply:
Better to use && to give a better idea. Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2088/commits/eac830d53d9f00ebefa78944532517deaeee7354